lean4-infoview/src/infoview/info.tsx

```diff
@@ -123,8 +124,37 @@ export function InfoDisplay(props0: InfoDisplayProps) {
     const hasGoals = status !== 'error' && goals;
     const hasTermGoal = status !== 'error' && termGoal;
     const hasMessages = status !== 'error' && messages.length !== 0;
-    const filterClasses = 'link pointer mh2 dim codicon fr ' + (reverseOrder ? 'codicon-arrow-up' : 'codicon-arrow-down');
-    const sortButton = <a className={filterClasses} onClick={e => { setReverseOrder(!reverseOrder); }} title="reverse list"/>
+    const sortClasses = 'link pointer mh2 dim codicon fr ' + (goalFilters.reverse ? 'codicon-arrow-up' : 'codicon-arrow-down');
+    const sortButton = <a className={sortClasses} title="reverse list" onClick={e => {
+        setGoalFilters(s => {
+            return { reverse: !s.reverse, isType: s.isType, isInstance: s.isInstance }
+        } ); }
+    } />
+
+    const filterMenu = <span>
+        <a className='link pointer' onClick={e => {
+            setGoalFilters(s => {
+                return { reverse: s.reverse, isType: !s.isType, isInstance: s.isInstance }
+            } ); }}>
+                <span className={'filterMenuIcon codicon ' + (goalFilters.isType ? 'codicon-eye' : 'codicon-eye-closed')}>&nbsp;</span>
```

Comment:
The "eye" codicon looks very cryptic to me.  Maybe it would be more intuitive if this was an actual checkbox, and the text was "show instances" (or "hide instances").

Reply:
Ok, I found another filter menu in the Problems UI that does use checkboxes, so consistency wins:
![image](https://user-images.githubusercontent.com/18707114/169890006-78cd2e8b-f517-4488-964b-5f2524bd9757.png)

Even though this is exactly what the eye icon was invented for :-) It literally means "show it so I can see it" or "hide it so I cannot see it", but I guess not enough apps have used it over the years which results in your reaction...

Problem is now I have to find a "blank filler" that is exactly the same size as the checkmark icon but codeicons has no such blank icon defined...

Reply:
I can't use a fixed pixel sized filler because it changes with vs code font size (Ctrl + and Ctrl -)

Reply:
I understand the eye icon in general, it's just that the codicons one doesn't look at all like an eye at all.  The checkmark is certainly clearer (at least to me).

> Problem is now I have to find a "blank filler" that is exactly the same size as the checkmark icon but codeicons has no such blank icon defined...

The two eye icons also had different sizes, it's not the end of the world if they don't match exactly.

Reply:
Turns out fixed pixel icon does work, the Ctrl+ key must be scaling the DPI.  I updated the demo video to show:

![image](https://user-images.githubusercontent.com/18707114/169914292-8db9a100-e446-44a0-b1d7-e5ab58228c59.png)

Reply:
Note: I did find one eye icon usage in vscode for lazily expanded variables in the debugger:
![image](https://user-images.githubusercontent.com/18707114/169933458-ba99b3d2-ce47-4650-9218-c700ccec7151.png)